src/impl/dtlstransport.cpp

```diff
@@ -434,13 +434,20 @@ DtlsTransport::DtlsTransport(shared_ptr<IceTransport> lower, certificate_ptr cer
 		SSL_set_options(mSsl, SSL_OP_SINGLE_ECDH_USE);
 		SSL_set_tmp_ecdh(mSsl, ecdh.get());
 
-		// RFC 8827: The DTLS-SRTP protection profile SRTP_AES128_CM_HMAC_SHA1_80 MUST be supported
-		// See https://www.rfc-editor.org/rfc/rfc8827.html#section-6.5 Warning:
+#if USE_GNUTLS
+		const char *srtpProfiles = "SRTP_AES128_CM_SHA1_80";
+#else
+		// Try to use GCM suite
+		const char *srtpProfiles = "SRTP_AEAD_AES_256_GCM:SRTP_AEAD_AES_128_GCM:SRTP_AES128_CM_SHA1_80";
+#endif
 		// SSL_set_tlsext_use_srtp() returns 0 on success and 1 on error
-		if (SSL_set_tlsext_use_srtp(mSsl, "SRTP_AES128_CM_SHA1_80"))
-			throw std::runtime_error("Failed to set SRTP profile: " +
-			                         openssl::error_string(ERR_get_error()));
-
+		if (SSL_set_tlsext_use_srtp(mSsl, srtpProfiles)) {
+			// RFC 8827: The DTLS-SRTP protection profile SRTP_AES128_CM_HMAC_SHA1_80 MUST be supported
+			// See https://www.rfc-editor.org/rfc/rfc8827.html#section-6.5 Warning:
```

Comment:
This means the profile must be supported by the remote peer, not by OpenSSL, so it would be better to keep it on top of the whole block to explain why SRTP_AES128_CM_SHA1_80 is the baseline.